src/mono/wasm/debugger/DebuggerTestSuite/DebuggerTestBase.cs

```diff
@@ -451,15 +475,17 @@ internal async Task<Result> SetValueOnObject(JToken obj, string property, string
                 locals_fn: locals_fn);
         }
 
-        internal async Task<JObject> EvaluateAndCheck(string expression, string script_loc, int line, int column, string function_name,
-            Func<JObject, Task> wait_for_event_fn = null, Action<JToken> locals_fn = null) => await SendCommandAndCheck(
-            JObject.FromObject(new { expression = expression }),
-            "Runtime.evaluate", script_loc, line, column, function_name,
-            wait_for_event_fn: wait_for_event_fn,
-            locals_fn: locals_fn);
+        internal async Task<JObject> EvaluateAndCheck(
+                                        string expression, string script_loc, int line, int column, string function_name,
+                                        Func<JObject, Task> wait_for_event_fn = null, Func<JToken, Task> locals_fn = null)
+            => await SendCommandAndCheck(
+                        JObject.FromObject(new { expression = expression }),
+                        "Runtime.evaluate", script_loc, line, column, function_name,
+                        wait_for_event_fn: wait_for_event_fn,
+                        locals_fn: locals_fn);
 
         internal async Task<JObject> SendCommandAndCheck(JObject args, string method, string script_loc, int line, int column, string function_name,
-            Func<JObject, Task> wait_for_event_fn = null, Action<JToken> locals_fn = null, string waitForEvent = Inspector.PAUSE)
+            Func<JObject, Task> wait_for_event_fn = null, Func<JToken, Task> locals_fn = null, string waitForEvent = Inspector.PAUSE)
```

Comment:
What do you think about adding two overloads each of `EvaluateAndCheck` and `SendCommandAndCheck`:  one that takes a `Action<JToken>` and one that takes a `Func<JToken, Task>` ?  That way most of the callers don't have to do the `locals_fn: async (l) => { ... await Task.CompletedTask }` change.

Reply:
IIRC, the problem with that was the default value, because of which if locals_fn wasn't passed, then it couldn't figure out which one to resolve.

(Typing on mobile)